### PR TITLE
General improvements to particles

### PR DESCRIPTION
**Pull request by Generalcamo**
Basically in this pull I changed the properties of the sound for Zaps to support pitch and volume bending, and also added a sound for Firestone Ore.

In addition, I added smoke particles to Firestone Ore's fire fountain.

### CODE REVIEW

File: src/main/java/mods/railcraft/client/particles/ParticleFireSpark.java [hunk 1]
```diff
@@ -100,6 +101,16 @@ public void onUpdate() {
...
 
+        /**
+         * Called to spawn smoke particles with the entity
+         */
+        float f = (float)this.particleAge / (float)this.particleMaxAge;
+
+        if (this.rand.nextFloat() > f)
+        {
+            this.worldObj.spawnParticle(EnumParticleTypes.SMOKE_NORMAL, this.posX, this.posY, this.posZ, this.motionX, this.motionY, this.motionZ, 0);
+        }
```
Comment (liach) [inline src/main/java/mods/railcraft/client/particles/ParticleFireSpark.java:112]: 😞

Comment (CovertJaguar) [reply src/main/java/mods/railcraft/client/particles/ParticleFireSpark.java:112]: I did think this was a little strange, I'll need to check what it does to the recharge animation too.

Comment (liach) [reply src/main/java/mods/railcraft/client/particles/ParticleFireSpark.java:112]: code style and javadoc!

Comment (CovertJaguar) [reply src/main/java/mods/railcraft/client/particles/ParticleFireSpark.java:112]: oh that...

**Pull request merged**